pkg/bearertoken/grpc_test.go

```diff
@@ -0,0 +1,297 @@
+				if test.wantToken == "" {
+					assert.False(t, ok, "expected no token")
+				} else {
+					assert.True(t, ok, "expected token to be present")
+					assert.Equal(t, test.wantToken, token)
+				}
+				return nil, nil
+			}
+
+			_, err := unaryInterceptor(test.ctx, nil, &grpc.UnaryServerInfo{}, unaryHandler)
+			if test.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedErr)
+			}
+
+			// Test stream server interceptor
+			streamInterceptor := NewStreamServerInterceptor()
+			mockStream := &mockServerStream{ctx: test.ctx}
+			streamHandler := func(_ any, stream grpc.ServerStream) error {
+				token, ok := GetBearerToken(stream.Context())
+				if test.wantToken == "" {
+					assert.False(t, ok, "expected no token")
+				} else {
+					assert.True(t, ok, "expected token to be present")
+					assert.Equal(t, test.wantToken, token)
+				}
+				return nil
+			}
+
+			err = streamInterceptor(nil, mockStream, &grpc.StreamServerInfo{}, streamHandler)
+			if test.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedErr)
+			}
+		})
+	}
+}
+
+func TestClientUnaryInterceptorWithBearerToken(t *testing.T) {
+	interceptor := NewUnaryClientInterceptor()
+
+	// Mock invoker
+	invoker := func(ctx context.Context, _ string, _ any, _ any, _ *grpc.ClientConn, _ ...grpc.CallOption) error {
+		md, ok := metadata.FromOutgoingContext(ctx)
+		require.True(t, ok)
+		assert.Equal(t, "test-token", md[Key][0])
+		return nil
+	}
+
+	// Context with token
+	ctx := ContextWithBearerToken(context.Background(), "test-token")
+
+	err := interceptor(ctx, "method", nil, nil, nil, invoker)
+	require.NoError(t, err)
+}
+
+func TestClientStreamInterceptorWithBearerToken(t *testing.T) {
+	interceptor := NewStreamClientInterceptor()
+
+	// Mock streamer
+	streamer := func(ctx context.Context, _ *grpc.StreamDesc, _ *grpc.ClientConn, _ string, _ ...grpc.CallOption) (grpc.ClientStream, error) {
+		md, ok := metadata.FromOutgoingContext(ctx)
+		require.True(t, ok)
+		assert.Equal(t, "test-token", md[Key][0])
+		return nil, nil
+	}
+
+	// Context with token
+	ctx := ContextWithBearerToken(context.Background(), "test-token")
+
+	_, err := interceptor(ctx, &grpc.StreamDesc{}, nil, "method", streamer)
+	require.NoError(t, err)
+}
+
+func TestServerUnaryInterceptorWithBearerToken(t *testing.T) {
+	interceptor := NewUnaryServerInterceptor()
+	testToken := "test-token"
+
+	// Test with token in context
+	handler := func(ctx context.Context, _ any) (any, error) {
+		token, ok := GetBearerToken(ctx)
+		require.True(t, ok)
+		assert.Equal(t, testToken, token)
+		return nil, nil
+	}
+
+	ctx := ContextWithBearerToken(context.Background(), testToken)
+	_, err := interceptor(ctx, nil, &grpc.UnaryServerInfo{}, handler)
+	require.NoError(t, err)
+}
+
+func TestServerStreamInterceptorWithBearerToken(t *testing.T) {
```

Comment:
isn't this already tested in `TestServerInterceptors`? Let's not duplicate the tests.